src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -6927,6 +6927,9 @@ static bool hasNoNonNullableCounterpart(TypeSymbol? type)
             // Note: the state may be conditional if a previous argument involved a conditional post-condition
             // The WhenTrue/False states correspond to the invocation returning true/false
 
+            // adjust parameter annotations based on presence of NotNullIfNotNullAttribute and state of other parameters
+            parameterAnnotations = notNullBasedOnParameters(parameterAnnotations, notNullParametersOpt, parameter);
+
```

Comment:
Perhaps this should be treated as a pre-condition for by value or `in` arguments instead? I.e. we know the state based on other arguments even before the invocation occurred.

Reply:
I think it's more straightforward and consistent to treat as a postcondition, as we do for other kinds of NotNullIfNotNull parameters. This is also what we do for other nullability attributes which can be applied to either by-value or by-ref parameters, such as NotNull and NotNullWhen ([SharpLab](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEAMACdBGAOgBEoIBzGAewGcAXKAY2oIGFKEBTAQRggBsAntSjUA3AFgAUFPQAmbCykBvKdjW4AzLjwA2bMEqU+2AJLUAcpVrmArnz4AKANqXrdvgHUAFhxgPaAE42HACUALramAD82AAOIdgAvAB82LReAZQA7tgw7gCEEpIAvjLyRMqq6vh66AAs2ACyeA740XEhVWoqkup92FAAZtgOLARmrrb2DvGdvf3dXQvqsQQAKpQAyoGwZA4hRcvYpfMLS32rG9sBu/uH6ifFQA===)).